src/Transform/ONNX/InstrumentONNXPass.cpp

```diff
@@ -36,33 +37,51 @@ namespace {
  * This pass insert KrnlInstrumentOp before and after each ONNX ops
  */
 
-// Strong-typed enum is NOT used because the value will be static_cast to int
-enum InstrumentActions {
-  InstrumentBeforeOp,
-  InstrumentAfterOp,
-  InstrumentReportTime,
-  InstrumentReportMemory
-};
-
-// Inherited issue: no default value support for cl::bits
-llvm::cl::bits<InstrumentActions> InstrumentControlBits(
-    llvm::cl::desc("Specify what instrumentation actions at runtime:"),
-    llvm::cl::values(
-        clEnumVal(InstrumentBeforeOp, "insert instrument before op"),
-        clEnumVal(InstrumentAfterOp, "insert instrument after op"),
-        clEnumVal(
-            InstrumentReportTime, "instrument runtime reports time usage"),
-        clEnumVal(
-            InstrumentReportMemory, "instrument runtime reports memory usage")),
-    llvm::cl::cat(onnx_mlir::OMPassOptions));
-
 class InstrumentONNXPass
     : public mlir::PassWrapper<InstrumentONNXPass, OperationPass<FuncOp>> {
 
+public:
+  Option<std::string> instrumentONNXOps{*this, "instrument-onnx-ops",
+      llvm::cl::desc("Specify onnx ops to be instrumented\n"
+                     "\"NONE\" or \"\" for no instrument\n"
+                     "\"ALL\" for all ops. \n"
+                     "\"op1 op2 ...\" for the specified ops."),
+      llvm::cl::init("")};
+
+  Option<bool> instrumentBefore{*this, "instrument-before",
+      llvm::cl::desc("insert instrument before op"), llvm::cl::init(false)};
+
+  Option<bool> instrumentAfter{*this, "instrument-after",
+      llvm::cl::desc("insert instrument after op"), llvm::cl::init(false)};
+
+  Option<bool> reportTime{*this, "report-time",
+      llvm::cl::desc("instrument runtime reports time usage"),
+      llvm::cl::init(false)};
+
+  Option<bool> reportMemory{*this, "report-memory",
+      llvm::cl::desc("instrument runtime reports memory usage"),
+      llvm::cl::init(false)};
+
+  InstrumentONNXPass() = default;
+  InstrumentONNXPass(const InstrumentONNXPass &pass)
+      : mlir::PassWrapper<InstrumentONNXPass, OperationPass<FuncOp>>() {
+    this->instrumentONNXOps = pass.instrumentONNXOps;
+    this->instrumentBefore = pass.instrumentBefore;
+    this->instrumentAfter = pass.instrumentAfter;
+    this->reportTime = pass.reportTime;
+    this->reportMemory = pass.reportMemory;
```

Comment:
Suggest to initialize these in the constructor init list:
```
  InstrumentONNXPass(const InstrumentONNXPass &pass)
      : mlir::PassWrapper<InstrumentONNXPass, OperationPass<FuncOp>>(),  instrumentONNXOps(pass.instrumentONNXOps), .... {
```
{

Reply:
removed initialization of options from the copy constructor.